misk-aws/src/main/kotlin/misk/dynamodb/DyEntity.kt

```diff
@@ -0,0 +1,14 @@
+package misk.dynamodb
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute
+
+/*
+ * This version field is used for optimistic locking.
+ */
+@DynamoDBTable(tableName = "")
+abstract class DyEntity {
```

Comment:
I would not use inheritance to model dynamo db tables because:
- it is difficult to gather a set of good defaults at this point
  - not every table wants a version field: declaring a version field prevents you from using condition expressions in write transactions. [reference](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.OptimisticLocking.html)
- it is not straightforward to override what the parent class customizes

Reply:
Agreed. Prefer interfaces over abstract classes.

Reply:
I've deleted this entity, but think I should keep the `DyTimeStampedEntity`.

 At first there was more code associated with making them work. Now its more simple (2 annotations on each member). Unfortunately if its an interface, I cannot have the attributes on the members; basically defeating the purpose IMO (know how to use these attributes is the tricky part).

If we're against abstract classes, then I will also kill the `DyTimeStampedEntity`

Reply:
Yeah I think we avoid using abstract classes when possible. 

BTW do we want to prefix the table class like we do with hibernate classes?